### PR TITLE
[requests] Enabled by default

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
This PR continues on from #658, enabling the requests integration by default.

In order to properly test the change a test was added to test the idempotence of patching requests when autopatching.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -542,11 +542,13 @@ jobs:
     steps:
       - checkout
       - *restore_cache_step
-      - run: tox -e 'requests_contrib-{py27,py34,py35,py36}-requests{208,209,210,211,212,213,219}' --result-json /tmp/requests.results
+      - run: tox -e 'requests_contrib-{py27,py34,py35,py36}-requests{208,209,210,211,212,213,219}' --result-json /tmp/requests.1.results
```
Comment (brettlangdon) [inline .circleci/config.yml]: I think you can do `'requests_contrib{,_autopatch}-{...'`

File: tox.ini [hunk 2]
```diff
@@ -318,6 +318,7 @@ commands =
     redis_contrib: nosetests {posargs} tests/contrib/redis
     rediscluster_contrib: nosetests {posargs} tests/contrib/rediscluster
     requests_contrib: nosetests {posargs} tests/contrib/requests
+    requests_contrib_autopatch: python tests/ddtrace_run.py nosetests {posargs} tests/contrib/requests/test_requests_autopatch.py
```
Comment (brettlangdon) [inline tox.ini]: `s/nosetests/pytest/`... eh?

Comment (brettlangdon) [reply tox.ini]: :)

Comment (Kyle-Verhoog) [reply tox.ini]: coming soon™

File: .circleci/config.yml (hunk 1 above)
Comment (Kyle-Verhoog) [reply .circleci/config.yml]: Indeed. @labbati is there a reason we have the other auto-patch tests separated out?

Comment (labbati) [reply .circleci/config.yml]: They can be unified now. The reason used to be that for autopatch we needed to dist and install while for the other tests we did not (to go faster). Now (after https://github.com/DataDog/dd-trace-py/pull/654) we do not need the "dist -> install" anymore and we should unify them (when possible).

Comment (Kyle-Verhoog) [reply .circleci/config.yml]: I'll leave it for this PR and open up a PR to clean them up.

**Pull request merged**